py-polars/src/series/export.rs

```diff
@@ -279,23 +304,41 @@ where
     let np_arr = PyArray1::from_iter_bound(py, ca.iter().map(mapper));
     np_arr.into_py(py)
 }
-/// Convert dates directly to i64 with i64::MIN representing a null value
+/// Convert booleans to u8 if no nulls are present, otherwise convert to objects.
+fn boolean_series_to_numpy(py: Python, s: &Series) -> PyObject {
+    let ca = s.bool().unwrap();
+    if s.null_count() == 0 {
+        let values = ca.into_no_null_iter();
+        PyArray1::<bool>::from_iter_bound(py, values).into_py(py)
+    } else {
+        let values = ca.into_iter().map(|opt_v| opt_v.into_py(py));
```

Comment:
We should use `iter`, that's faster.

Reply:
Yeah I was actually wondering why we used `into_iter` here, but I figured I'd look at it later. Simply using `iter` instead just works though 👍